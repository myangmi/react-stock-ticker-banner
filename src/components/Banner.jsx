import React, { useEffect, useState } from 'react';
import PropTypes, { string } from 'prop-types';
import axios from 'axios';
import Tickers from './Tickers.jsx';

import './Banner.scss';

const propTypes = {
  tickers: PropTypes.arrayOf(string).isRequired,
  apiKey: PropTypes.string,
};

const defaultProps = {
  apiKey: 'Tpk_f4c74570f4f44b0facf47abd0ba3ae4c', // TODO: Remove personal Sandbox Token
};

const Banner = ({ tickers, apiKey }) => {
  const [quoteData, setQuoteData] = useState({});
  const apiPrefix = (apiKey[0] === 'T') ? 'https://sandbox.iexapis.com' : 'https://cloud.iexapis.com';

  useEffect(() => {
    axios.get(
      `${apiPrefix}/stable/stock/market/batch?symbols=${tickers}&token=${apiKey}&types=price`,
    ).then(
      (res) => {
        setQuoteData(res.data);
      },
    ).catch((err) => {
      console.log(err);
    });
  }, []);

  return (
    <div className='Banner' >
      <Tickers quoteData={quoteData} />
    </div>
  );
};

Banner.defaultProps = defaultProps;
Banner.propTypes = propTypes;
export default Banner;
