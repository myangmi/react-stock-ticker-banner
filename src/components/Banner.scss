@import url('https://fonts.googleapis.com/css?family=Montserrat');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

:root {
  --banner-width: 100%;
  --banner-height: 100%;
  --banner-elements: 4;
  --banner-elements-displayed: 2;
  --banner-element-width: calc(var(--banner-width) / var(--banner-elements-displayed));
  --banner-animation-duration: calc(var(--banner-elements) * 3s);
}

body {
  font-family: 'Montserrat', sans-serif;
  background-color: #eee;
  color: #111;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}


.Banner {
  width: var(--banner-width);
  height: var(--banner-height);
  background-color: black;
  color: green;
  overflow: hidden;
}

@keyframes scrolling {
  0% {
    transform: translateX(0vw);
  }
  100% {
    transform: translateX(calc(-1 * var(--banner-element-width) * var(--banner-elements)));
  }
}

.Tickers {
  list-style: none;
  height: 100%;
  display: flex;
  animation: scrolling var(--banner-animation-duration) linear infinite;
}

.Tickers li {
  display: flex;
  justify-content: center;
  align-items: center;
  /* text-align: center; */
  flex-shrink: 0;
  width: var(--banner-element-width);
  max-height: 100%;
  font-size: calc(var(--banner-height)*3/4); /* 5rem; */
  white-space: nowrap;
}
