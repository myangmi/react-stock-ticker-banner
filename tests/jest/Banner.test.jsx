import React from 'react';
import { shallow, mount } from 'enzyme';

import Banner from '../../src/components/Banner.jsx';
import mockTickers from '../mocks/mockTickers';

describe('Banner', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Banner tickers={mockTickers} apiKey={'mockAPIKey'}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should set apiKey prop to default sandbox key if apiKey is not present', () => {
    const wrapper = mount(<Banner tickers={mockTickers} />);

    expect(wrapper.prop('apiKey')).toEqual('Tpk_f4c74570f4f44b0facf47abd0ba3ae4c');
  });
});
