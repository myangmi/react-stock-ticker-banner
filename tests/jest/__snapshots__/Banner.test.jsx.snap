// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should render properly 1`] = `
<div
  className="Banner"
>
  <Tickers
    quoteData={Object {}}
  />
</div>
`;
